k8scrdclient/k8scrdclient.go

```diff
@@ -91,20 +91,34 @@ func (c *CRDClient) ensureCreated(ctx context.Context, crd *apiextensionsv1beta1
 			return microerror.Mask(err)
 		}
 
-		for _, cond := range manifest.Status.Conditions {
-			switch cond.Type {
-			case apiextensionsv1beta1.Established:
-				if cond.Status == apiextensionsv1beta1.ConditionTrue {
-					return nil
-				}
-			case apiextensionsv1beta1.NamesAccepted:
-				if cond.Status == apiextensionsv1beta1.ConditionFalse {
-					return microerror.Maskf(nameConflictError, cond.Reason)
-				}
+		// In case the CRDs names are not accepted we have to stop processing here
+		// and return the reason of the failing condition. Therefore we stop retries
+		// permanently.
+		{
+			con, ok := statusCondition(manifest.Status.Conditions, apiextensionsv1beta1.NamesAccepted)
```

Comment:
Is there a way to look up which other conditions could be `fatal`?
Or is it only `NamesAccepted` and `NonStructuralSchema` and we are confident that this is enough.

Reply:
I think we got all relevant now. Check here good Sir: https://godoc.org/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1#CustomResourceDefinitionConditionType.